### PR TITLE
Refining advanced filter in patients page

### PR DESCRIPTION
**Pull request by sam9111**
- Added date range filter based on "Onset of Symptoms Date"
- Added district filter
- Fixed bug for clearing previous filters and some filters not being applied after changing it for the second time
- Added badges for some filters which were missing/incorrect before

### CODE REVIEW

File: src/Components/Patient/PatientFilterV2.tsx [hunk 1]
```diff
@@ -84,19 +87,60 @@ export default function PatientFilterV2(props: any) {
...
-        setFacilityLoading(false);
-      }
+      // if (filter.facility) {
+      //   setFacilityLoading(true);
+      //   const { data: facilityData } = await dispatch(
+      //     getFacility(filter.facility, "facility")
+      //   );
+      //   setFilterState({ facility_ref: facilityData });
+      //   setFacilityLoading(false);
+      // }
```
Comment (smartclash) [inline src/Components/Patient/PatientFilterV2.tsx]: Remove the code if not required

**Review by smartclash — CHANGES_REQUESTED** [review]

**Comment by samholics** [comment]
tested, failed

when you applied district filter on patients list ;
 . logged in as state  admin, and  selected a  district .but the patient count remains the same. not showing the results
  also the selected district cleared when we came back to filters

. date of onset symptoms filter is working fine

**Comment by sam9111** [comment]
The district filter is now working.

chrome-capture (1)

**Comment by amruthavijay88** [comment]
Tested, when filtered by a district - it works, but when we re-open advanced filters again - it doesn't show which filter is applied.

**Comment by amruthavijay88** [comment]
@sam9111

**Comment by amruthavijay88** [comment]
Tested - date range filters and district filters are working fine. Previous mentioned issue is resolved too.

**Pull request merged**